plugins/inputs/docker_log/docker_log.go

```diff
@@ -311,6 +316,37 @@ func (d *DockerLogs) tailContainerLogs(
 	}
 }
 
+func parseLogEntry(rawMessage []byte, acc *telegraf.Accumulator, containerID *string) (time.Time, []byte) {
+	var message []byte
+	switch length := len(rawMessage); {
```

Comment:
In this function use `bytes.SplitN(line, []byte(" "), 2)`. if there aren't 2 splits return an error, otherwise you are good to parse and return (time, message, err)

Reply:
Looks pretty good,  but I think theres is one edge case here,  when the message is empty, I mean that it is only <timestamp> without any trailing whitespace, it is not an error. So in this case we need to return ts, empty message, and nil error. The caller function will make a decision either emit empty message as a telegraf metric or not.
Moreover,  I think that we should emit empty messages, as that is what container produced, but right now,  this is thrown away. What is your opinion?

Reply:
Let's emit the empty string as the message.

Reply:
Updated as discussed